### DISCUSSION TITLE
Is there any way to change location of log file?

### DISCUSSION THREAD

**Discussion by hrkachhia** [theory/sqitch#740] [General]
Hello ,

I am getting below warning and error while deploy, revert and verify changes in docker sqitch:

```
**Failed to initialize log. No logging is enabled: [Errno 13] Permission denied: '/snowsql_rt.log_bootstrap'
We were unable to create or write to the ../snowsql_rt.log. Make sure you have permission to write to the log file's parent folder or to modify the location of the log file specified in the SnowSQL log_file configuration option. See docs: https://docs.snowflake.com/en/user-guide/snowsql-config.html#log-file**

**Observed error: [Errno 13] Permission denied: '/snowsql_rt.log'**
```

I have already modified location of log file in `~/.snowsql/config`  as below.  But Still, I am getting error and it is not creating a log file at provided location. 

``` ini
[connections]
accountname = my-snowflake-account
username = username
password = password
rolename =my-role
warehousename = warehouse
log_file =/home/mydir/Public/sqitch/my-project/log
log_bootstrap_file/home/mydir/Public/sqitch/my-project/log/bootlog
```

**Comment by theory** [comment]
How are you running Sqitch? If you're using the Docker image, you should use the [shell script](https://github.com/sqitchers/docker-sqitch/#linux-macos-git-bash) and set the `SQITCH_IMAGE` environment variable to the name if the image you built. That will mount your home directory.

**Reply by hrkachhia** [reply]
I am using Sqitch using shell script. I have added docker image name in the `docker-switch.sh` file but I haven't set the `SQITCH_IMAGE` environment variable. 

I have applied below changes in docker-switch.sh script : 

``` sh
# Determine which Docker image to run.
#SQITCH_IMAGE=${SQITCH_IMAGE:=sqitch/sqitch:v1.3.1-snowflake}

SQITCH_IMAGE=sqitch/sqitch:snowflake

# Handle OS-specific options.
case "$(uname -s)" in
    Linux*)
        passopt+=(-e "SQITCH_ORIG_FULLNAME=$(getent passwd $user | cut -d: -f5 | cut -d, -f1)")
      #  passopt+=(-u $(id -u ${user}):$(id -g ${user}))
        ;;
    Darwin*)
        passopt+=(-e "SQITCH_ORIG_FULLNAME=$(/usr/bin/id -P $user | awk -F '[:]' '{print $8}')")
        ;;
    MINGW*|CYGWIN*)
        passopt+=(-e "SQITCH_ORIG_FULLNAME=$(net user $user)")
        ;;
    *)
        echo "Unknown OS: $(uname -s)"
        exit 2
        ;;
esac
```

**Reply by theory** [reply]
Probably, then, the Docker container does not have write access to your home directory for log files. What happens if you remove the `log_file` option?

**Reply by theory** [reply]
Also it looks like you're missing an equal sign after `log_bootstrap_file`.

**Reply by hrkachhia** [reply]
I have removed the log_file option from the config file. Still, getting the same error. 
Which access rights  require for docker container? 

When I directly execute below command on console I am getting error as below. 

```
 docker run --rm --network host \                                                                                                                             
             --mount "type=bind,src=$(pwd),dst=/repo" \
             --mount "type=bind,src=$HOME,dst=$homedst" \
             "${passopt[@]}" "$SQITCH_IMAGE" "$@"
```
**fish: ${ is not a valid variable in fish.
    "${passopt[@]}" "$SQITCH_IMAGE" "$@"**

**Comment by theory** [comment]
What OS are you running this on? Linux?

**Reply by hrkachhia** [reply]
I am using Ubuntu 20.04.4 LTS x86_64

**Reply by theory** [reply]
So, something is off, such that your home directory mounted inside the container cannot be written to.

**Reply by hrkachhia** [reply]
Okay then what's changes require to works log file? Could you please suggest?

**Comment by theory** [comment]
I am not sure. Try some of the suggestions in [this SO question](https://stackoverflow.com/q/39397548/79202).

**Reply by theory** [reply]
Or just comment out those lines and don't log stuff.
